mercury/mercury-library/vc/src/main/scala/io/iohk/atala/mercury/vc/dif/ProposeCredential.scala

```diff
@@ -0,0 +1,70 @@
+package io.iohk.atala.mercury.vc.dif
+
+type DID = String
+
+case class CredentialManifest(
+    issuer: DID,
+    credential: NameAndSchema,
+)
+
+case class NameAndSchema(
+    name: String,
+    schema: String,
+)
+
+///////////////////////////////////////////////////////////
+
+object ProposeCredentialFormat {
+  def formatDIF = "dif/credential-manifest@v1.0"
+}
+
+/** @see
+  *   https://github.com/hyperledger/aries-rfcs/blob/main/features/0511-dif-cred-manifest-attach/README.md#propose-credential-attachment-format
+  */
+final case class ProposeCredentialFormat(
+    issuer: DID,
+    credential: NameAndSchema,
+)
+
+///////////////////////////////////////////////////////////
+
+object OfferCredentialFormat {
+  def formatDIF = "dif/credential-manifest@v1.0"
+}
+
+/** @see
+  *   https://github.com/hyperledger/aries-rfcs/blob/main/features/0511-dif-cred-manifest-attach/README.md#offer-credential-attachment-format
+  */
+final case class OfferCredentialFormat(
+    challenge: String,
+    domain: String,
+    credential_manifest: CredentialManifestObject,
+)
+
+type CredentialManifestObject = Any
```

Comment:
Is it possible to define the set of possible types here?

Reply:
Let's see what @FabioPinheiro thinks about this, but do we spend time on this, knowing that we do not support the `propose` phase for now?